### PR TITLE
Update dynamic filtering documentation for IS NOT DISTINCT FROM

### PR DESCRIPTION
**Pull request by raunaqmorarka**

### CODE REVIEW

File: presto-docs/src/main/sphinx/admin/dynamic-filtering.rst [hunk 1]
```diff
@@ -211,6 +213,8 @@ Limitations
 * Dynamic filtering is currently implemented only for :doc:`/connector/hive` and :doc:`/connector/memory` connectors.
 * Push down of dynamic filters into local table scan on worker nodes is limited to broadcast joins.
 * Min-max dynamic filter collection is not supported for DOUBLE, REAL and unorderable data types.
+* When ``IS NOT DISTINCT FROM`` is used in a join condition, dynamic filtering is not
+  supported if DOUBLE or REAL data types are part of the comparison condition.
```
Comment (sopel39) [inline presto-docs/src/main/sphinx/admin/dynamic-filtering.rst:217]: `are part of the comparison condition` -> `are used in the comparison expression`

File: presto-docs/src/main/sphinx/admin/dynamic-filtering.rst [hunk 2]
```diff
@@ -46,7 +46,9 @@ Analysis and confirmation
 Dynamic filtering depends on a number of factors:
 
 * Planner support for dynamic filtering for a given join operation in Presto.
-  Currently inner and right joins with comparison join conditions and semi-joins with IN conditions are supported.
+  Currently inner and right joins with ``=``, ``<``, ``<=``, ``>``, ``>=`` or
+  ``IS NOT DISTINCT FROM`` join conditions and semi-joins
+  with IN conditions are supported.
```
Comment (sopel39) [inline presto-docs/src/main/sphinx/admin/dynamic-filtering.rst:51]: inline with previous line

File: presto-docs/src/main/sphinx/admin/dynamic-filtering.rst [hunk 3]
```diff
@@ -46,7 +46,9 @@ Analysis and confirmation
 Dynamic filtering depends on a number of factors:
 
 * Planner support for dynamic filtering for a given join operation in Presto.
-  Currently inner and right joins with comparison join conditions and semi-joins with IN conditions are supported.
+  Currently inner and right joins with ``=``, ``<``, ``<=``, ``>``, ``>=`` or
+  ``IS NOT DISTINCT FROM`` join conditions and semi-joins
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/admin/dynamic-filtering.rst:50]: conditions, and semi-joins

File: presto-docs/src/main/sphinx/admin/dynamic-filtering.rst (hunk 2 above)
Comment (mosabua) [inline presto-docs/src/main/sphinx/admin/dynamic-filtering.rst:51]: apostrophes around IN

File: presto-docs/src/main/sphinx/admin/dynamic-filtering.rst (hunk 1 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/admin/dynamic-filtering.rst:217]: quote DOUBLE and REAL if you uppercase them

**Review by mosabua — CHANGES_REQUESTED** [review]
After minor feedback is address this looks good.

File: presto-docs/src/main/sphinx/admin/dynamic-filtering.rst (hunk 2 above)
Comment (raunaqmorarka) [reply presto-docs/src/main/sphinx/admin/dynamic-filtering.rst]: There are doc guidelines to stay within 80 chars, it doesn't affect the rendering of the line in the documentation.

**Review by mosabua — APPROVED** [review]

File: presto-docs/src/main/sphinx/admin/dynamic-filtering.rst [hunk 4]
```diff
@@ -46,7 +46,9 @@ Analysis and confirmation
 Dynamic filtering depends on a number of factors:
 
 * Planner support for dynamic filtering for a given join operation in Presto.
-  Currently inner and right joins with comparison join conditions and semi-joins with IN conditions are supported.
+  Currently inner and right joins with ``=``, ``<``, ``<=``, ``>``, ``>=`` or
+  ``IS NOT DISTINCT FROM`` join conditions, and
+  semi-joins with ``IN`` conditions are supported.
```
Comment (findepi) [inline presto-docs/src/main/sphinx/admin/dynamic-filtering.rst:51]: do we have semi-joins with a condition other than `IN`?

File: presto-docs/src/main/sphinx/admin/dynamic-filtering.rst [hunk 5]
```diff
@@ -210,7 +212,9 @@ Limitations
 
 * Dynamic filtering is currently implemented only for :doc:`/connector/hive` and :doc:`/connector/memory` connectors.
 * Push down of dynamic filters into local table scan on worker nodes is limited to broadcast joins.
-* Min-max dynamic filter collection is not supported for DOUBLE, REAL and unorderable data types.
+* Min-max dynamic filter collection is not supported for ``DOUBLE``, ``REAL`` and unorderable data types.
+* When ``IS NOT DISTINCT FROM`` is used in a join condition, dynamic filtering is not
+  supported if ``DOUBLE`` or ``REAL`` data types are used in the comparison expression.
```
Comment (findepi) [inline presto-docs/src/main/sphinx/admin/dynamic-filtering.rst:217]: "Dynamic filtering is not supported for `DOUBLE` and `REAL` data types when using `IS NOT DISTINCT FROM` predicate."

?

File: presto-docs/src/main/sphinx/admin/dynamic-filtering.rst [hunk 6]
```diff
@@ -46,7 +46,9 @@ Analysis and confirmation
 Dynamic filtering depends on a number of factors:
 
 * Planner support for dynamic filtering for a given join operation in Presto.
-  Currently inner and right joins with comparison join conditions and semi-joins with IN conditions are supported.
+  Currently inner and right joins with ``=``, ``<``, ``<=``, ``>``, ``>=`` or
+  ``IS NOT DISTINCT FROM`` join conditions, and
```
Comment (findepi) [inline presto-docs/src/main/sphinx/admin/dynamic-filtering.rst:50]: What about `IS DISTINCT FROM` (without NOT)?

File: presto-docs/src/main/sphinx/admin/dynamic-filtering.rst (hunk 4 above)
Comment (raunaqmorarka) [reply presto-docs/src/main/sphinx/admin/dynamic-filtering.rst:51]: that's to clarify that dynamic filtering with `NOT IN` is not supported (anti-joins)

File: presto-docs/src/main/sphinx/admin/dynamic-filtering.rst (hunk 6 above)
Comment (raunaqmorarka) [reply presto-docs/src/main/sphinx/admin/dynamic-filtering.rst:50]: We haven't added DF support for 'not-equals` and `IS DISTINCT FROM` yet

File: presto-docs/src/main/sphinx/admin/dynamic-filtering.rst (hunk 5 above)
Comment (raunaqmorarka) [reply presto-docs/src/main/sphinx/admin/dynamic-filtering.rst]: updated

File: presto-docs/src/main/sphinx/admin/dynamic-filtering.rst [hunk 7]
```diff
@@ -210,7 +212,8 @@ Limitations
 
 * Dynamic filtering is currently implemented only for :doc:`/connector/hive` and :doc:`/connector/memory` connectors.
 * Push down of dynamic filters into local table scan on worker nodes is limited to broadcast joins.
-* Min-max dynamic filter collection is not supported for DOUBLE, REAL and unorderable data types.
+* Min-max dynamic filter collection is not supported for ``DOUBLE``, ``REAL`` and unorderable data types.
+* Dynamic filtering is not supported for DOUBLE and REAL data types when using IS NOT DISTINCT FROM predicate.
```
Comment (sopel39) [inline presto-docs/src/main/sphinx/admin/dynamic-filtering.rst:216]: We should wrap `DOUBLE` and `REAL` with "``" for consistency

**Review by sopel39 — APPROVED** [review]

**Review by sopel39 — APPROVED** [review]

File: presto-docs/src/main/sphinx/admin/dynamic-filtering.rst [hunk 8]
```diff
@@ -210,7 +212,8 @@ Limitations
 
 * Dynamic filtering is currently implemented only for :doc:`/connector/hive` and :doc:`/connector/memory` connectors.
 * Push down of dynamic filters into local table scan on worker nodes is limited to broadcast joins.
-* Min-max dynamic filter collection is not supported for DOUBLE, REAL and unorderable data types.
+* Min-max dynamic filter collection is not supported for ``DOUBLE``, ``REAL`` and unorderable data types.
+* Dynamic filtering is not supported for ``DOUBLE`` and ``REAL`` data types when using IS NOT DISTINCT FROM predicate.
```
Comment (findepi) [inline presto-docs/src/main/sphinx/admin/dynamic-filtering.rst:216]: put IS NOT DISTINCT FROM in backticks

**Review by findepi — APPROVED** [review]

**Pull request merged**